OpenRA.Mods.Common/Widgets/Logic/MissionBrowserLogic.cs

```diff
@@ -300,14 +303,32 @@ void StartMissionClicked()
 				PlayVideo(fsPlayer, gameStartVideo, PlayingVideo.GameStart, () =>
 				{
 					StopVideo(fsPlayer);
-					Game.StartMission(selectedMapPreview.Uid, gameSpeed, difficulty, onStart);
+					StartMission();
 				});
 			}
 			else
-				Game.StartMission(selectedMapPreview.Uid, gameSpeed, difficulty, onStart);
-		}
+				StartMission();
+        }
+
+		void StartMission()
+		{
+			OrderManager om = null;
+
+			Action lobbyReady = null;
+			lobbyReady = () =>
```

Comment:
You should be able to generalize this for both cases by having it take a list of orders to issue as arguments (with gamespeed/difficulty for missions and sync_lobby for skirmish)